src/ORM/Association/Loader/SelectLoader.php

```diff
@@ -294,13 +294,14 @@ protected function _addFilteringJoin($query, $key, $subquery)
         }
         $subquery->select($filter, true);
 
+        $conditions = null;
         if (is_array($key)) {
             $conditions = $this->_createTupleCondition($query, $key, $filter, '=');
         } else {
             $filter = current($filter);
         }
 
-        $conditions = isset($conditions) ? $conditions : $query->newExpr([$key => $filter]);
+        $conditions = $conditions ?: $query->newExpr([$key => $filter]);
```

Comment:
Could conditions be defined (not null) but falsey?

Reply:
No, it can only be `null` or `TupleComparison` (returned by `_createTupleCondition()`).